### ISSUE TITLE
Disable on-page JS but allow userscripts

### ISSUE THREAD

**Issue by hseg** [qutebrowser/qutebrowser#8010]
Currently, with `content.javascript.enabled = false`, userscripts don't run either. Thanks to @toofar and @emanuele6 in IRC, TIL adding a `@qute-js-world` directive set to anything other than `main` will exclude userscripts from this restriction.
- This is underdocumented -- perhaps we need a page on supported Greasemonkey directives?
- Is there a reason for the script world being set to `main` as default? IIUC, all that buys you is access to the document's JS variables, which to me seems a niche purpose (though my request to have userscripts run but not content scripts might also be called niche). At least, it causes confusion in the settings, as here.

**Comment by toofar** [comment]
> Is there a reason for the script world being set to main as default?

There are scripts on greasyfork etc which expect to be able to access the same window object as the page. And all the polyfill ones. There is probably some discussion of it on a PR somewhere, but breaking support for scripts that work elsewhere is a no-go. Although I see voilentmonkey says it'll look at the `requires unsafeWindow` thing to see what a script needs.
Maybe we could add a setting to change the default world?

**Comment by The-Compiler** [comment]
I agree on the docs, I added an item to #3238 for that. I don't think there's anything else we should do here. It's a per-script decision whether a script needs to live in the main world or not. For the same reason, I'm also against a global setting for a default world, as I'm pretty sure it'd lead to hard to debug issues where someone changes the world and has some scripts break without much context why.

**Comment by hseg** [comment]
> > Is there a reason for the script world being set to main as default?
> 
> There are scripts on greasyfork etc which expect to be able to access the same window object as the page. And all the polyfill ones. There is probably some discussion of it on a PR somewhere, but breaking support for scripts that work elsewhere is a no-go.

Fair enough -- I didn't know this was the default for GM in general, retracted.
